x/inflation/keeper/msg_server.go

```diff
@@ -0,0 +1,26 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/evmos/evmos/v10/x/inflation/types"
+	"github.com/pkg/errors"
+)
+
+var _ types.MsgServer = &Keeper{}
+
+func (k Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	if k.authority != req.Authority {
+		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.authority, req.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := k.SetParams(ctx, req.Params); err != nil {
+		return nil, errors.Wrapf(err, "error setting params")
```

Comment:
Nit: Commit this change if above suggestion is committed
```suggestion
		return nil, errorsmod.Wrapf(err, "error setting params")
```